logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/selfDeletingMessages/SelfDeletionTimer.kt

```diff
@@ -22,30 +22,28 @@ import com.wire.kalium.util.serialization.toJsonElement
 import kotlin.time.Duration
 import kotlin.time.Duration.Companion.ZERO
 
-sealed class SelfDeletionTimer {
+sealed interface SelfDeletionTimer {
+    val duration: Duration?
```

Comment:
Isn't it redundant?

We already have `Disabled`.

If this is just a shortcut for not having to type-check for both `Enabled` and `Enforced`, I'd suggest:

```kotlin
sealed interface SelfDeletionTimer {

    object Disabled: SelfDeletionTimer

    sealed interface Enabled: SelfDeletionTimer {
        val duration: Duration

        data class ByChoice(override val duration: Duration) : Enabled
    }     

    sealed interface Enforced: Enabled {

        data class ByTeam(override val duration: Duration) : Enforced
        data class ByGroup(override val duration: Duration) : Enforced
    }
}
```

This way we can do:
```kotlin

val duration = if (timer is Enabled) {
    timer.duration
} ?: null

```

And maybe improve the readability with these types:
```kotlin
SelfDeletionTimer.Disabled
SelfDeletionTimer.Enabled.ByChoice(duration)
SelfDeletionTimer.Enforced.ByTeam(duration)
SelfDeletionTimer.Enforced.ByGroup(duration)
```

Reply:
Marking as resolved as I see that `Enabled` should have nullable durability as well.